### DISCUSSION TITLE
Response validation

### DISCUSSION THREAD

**Discussion by lyda** [oapi-codegen/oapi-codegen#1196] [Q&A] [answered]
Is there any appetite for a response validation system for the default echo server? I assume this could be done via middleware.

This seems like it would be useful - it would have prevented an annoying error in some APIs we were developing.  But I'm wondering if there are difficulties in implementing this.

**Comment by jamietanna** [comment, marked as answer]
Yep, this can be done using the middleware, rather than in the code this project implements.

If you want to catch this in tests, you can also use https://pkg.go.dev/openapi.tanna.dev/go/validator#section-readme
